### PR TITLE
Go mod sdk update

### PR DESCRIPTION
**Pull request by carlossscastro**
Update Go to 1.16
Migrate to Go Modules
Update to Infra SDK v3.6.7

### CODE REVIEW

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -9,25 +9,20 @@ on:
...
       - uses: actions/checkout@v2
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
-          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
-      - name: Validate code
-        run: make ci/validate
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
```
Comment (gsanchezgavier) [inline .github/workflows/prerelease.yml:25]:
```suggestion
        uses: golangci/golangci-lint-action@v2
        with:
          only-new-issues: true
```

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -9,25 +9,20 @@ on:
...
   INTEGRATION: "apache"
   ORIGINAL_REPO_NAME: ${{ github.event.repository.full_name }}
   REPO_FULL_NAME: ${{ github.event.repository.full_name }}
   TAG: ${{ github.event.release.tag_name }}
 
 jobs:
-  validate:
-    name: Validate code via linters
-    runs-on: ubuntu-20.04
+  golangci-lint:
```
Comment (gsanchezgavier) [inline .github/workflows/prerelease.yml:19]: don't forget to update the conditions on the following jobs 
https://github.com/newrelic/nri-apache/pull/59/files#diff-884bb2e52646d4b226c75e5dd7c196f9d654accb529fef4b6eeb401daa83144eR102

File: .github/workflows/push_pr.yml [hunk 3]
```diff
@@ -11,23 +11,17 @@ env:
...
-      - name: Login to DockerHub
-        if: ${{env.DOCKER_LOGIN_AVAILABLE}}
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
-          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
-      - name: Validate code
-        run: make ci/validate
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
```
Comment (gsanchezgavier) [inline .github/workflows/push_pr.yml:24]: same here `only-new-issues: true`

File: tests/integration/helpers/helpers.go [hunk 4]
```diff
@@ -4,16 +4,14 @@ import (
...
 	"os/exec"
-	"reflect"
 	"strings"
-	"testing"
 )
 
 // GetTestName returns the name of the running test.
-func GetTestName(t *testing.T) interface{} {
-	v := reflect.ValueOf(*t)
-	return v.FieldByName("name")
```
Comment (gsanchezgavier) [inline tests/integration/helpers/helpers.go:15]: i think you can change this to `return t.Name()`

File: README.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ For installation and usage instructions, see our [documentation web site](https:
 
 ## Building
 
-Golang is required to build the integration. We recommend Golang 1.11 or higher.
+Golang is required to build the integration. We recommend Golang 1.16 or higher.
```
Comment (gsanchezgavier) [inline README.md:17]: could you remove the govendor [reference](https://github.com/newrelic/nri-apache/pull/59/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R39) too?

**Review by gsanchezgavier — COMMENTED** [review]
Thanks! looks good, i wrote some minor comments , also i think [this](https://github.com/newrelic/nri-redis/blob/master/build/release.mk#L24) is missing, if not the pre-release will fail.

File: .github/workflows/push_pr.yml (hunk 3 above)
Comment (carlossscastro) [reply .github/workflows/push_pr.yml:24]: 👍

File: .github/workflows/prerelease.yml (hunk 1 above)
Comment (carlossscastro) [reply .github/workflows/prerelease.yml:25]: Done. Thx!

File: .github/workflows/prerelease.yml (hunk 2 above)
Comment (gsanchezgavier) [reply .github/workflows/prerelease.yml:19]: line 107 of this file

**Review by gsanchezgavier — APPROVED** [review]
Thanks!

**Pull request merged**